dwa_local_planner/src/dwa_planner_ros.cpp

```diff
@@ -160,6 +166,14 @@ namespace dwa_local_planner {
         auto configuration = std::make_shared<DWAPlannerConfiguration>(name + "/" + mode_name);
         mode_configurations_.push_back(configuration);
       }
+
+      private_nh.param<std::string>( "mode_topic", mode_topic_, "mode" );
```

Comment:
I think the reconfigure callback might always get called by ros on initialization, so you don't have to publish the mode here separately.  It's worth testing.

Reply:
I am 85% sure that for some reason it does not, which is why I added this in the first place. I unfortunately probably don't have time to test it today.